shared/common-adapters/floating-menu.js

```diff
@@ -36,3 +36,42 @@ export default (props: Props) => {
     </FloatingBox>
   )
 }
+
+export type FloatingMenuParentProps = {
+  attachmentRef: ?React.Component<*, *>,
+  showingMenu: boolean,
+  setAttachmentRef: ?(?React.Component<*, *>) => void,
+  setShowingMenu: boolean => void,
+  toggleShowingMenu: () => void,
+}
+
+type State = {
+  attachmentRef: ?React.Component<*, *>,
+  showingMenu: boolean,
+}
+
+type Callbacks = {
+  setShowingMenu: boolean => void,
+  toggleShowingMenu: () => void,
+  setAttachmentRef: ?(?React.Component<*, *>) => void,
+}
+
+export const FloatingMenuParentHOC = <T: FloatingMenuParentProps>(
+  ComposedComponent: React.ComponentType<T>
+): React.ComponentType<$Diff<T, FloatingMenuParentProps>> => {
+  class FloatingMenuParent extends React.Component<$Diff<T, FloatingMenuParentProps>, State> {
+    _setters: Callbacks
+    constructor() {
```

Comment:
shouldn't this be:
```
constructor(props: $Diff<...>) {
  super(props)
}
```
?

Something always warns/yells at me whenever I forget to do it that way.